textile/features.textile

```diff
@@ -1510,43 +1508,52 @@ h4. ChannelMetrics
-*** @(BPA2b1)@ This error should only be set if it relates to a partial success. All fatal errors should be handled via language appropriate error handling.
-
-h4. BatchPublishResponse
-* @(BPB1)@ Contains information for each batch publish request within a @BatchResult@
-* @(BPB2)@ @BatchPublishResponse@ has the following attributes:
-** (@BPB2a)@ @channel@ is the channel name which this publish request was directed to
-** (@BPB2b)@ @messageId@ contains the resultant message ID, if the request succeeds and is null if @error@ is present
-** (@BPB2c)@ @error@ contains an @ErrorInfo@ object if this publish request failed, and is null if it succeeded
-
-h4. BatchPresenceResponse
-* @(BPD1)@ Contains information for each batch presence request within a @BatchResult@
-* @(BPD2)@ @BatchPresenceResponse@ contains the following attributes:
-** @(BPD2a)@ @channel@ is the channel name which this presence request
-** @(BPD2b)@ @presence@ is an array of presence data for the @channel@
-
-h4. BatchPresence
-* @(BPE1)@ Is a partial @PresenceMessage@ object containing @clientId@ and @action@, or @error@ if the presence failed
-* @(PBE2)@ This clause has been renamed to #BPE2.
-** @(PBE2a)@ This clause has been renamed to #BPE2a.
-** @(PBE2b)@ This clause has been renamed to #BPE2b.
-** @(PBE2c)@ This clause has been renamed to #BPE2c.
-* @(BPE2)@ @BatchPresence@ contains the following attributes:
-** @(BPE2a)@ @clientId@ - identical to #TP3c
-** @(BPE2b)@ @action@ - identical to #TP3b - null if @error@ is present
-** @(BPE2c)@ @error@ - an @ErrorInfo@ object representing the failure reason for this channel - null if @action@ is present
+h4. BatchPublishResult
+
+* @(BPR1)@ Contains information about the result of publishes requested by a single @BatchPublishSpec@
+* @(BPR2)@ The attributes of @BatchPublishResult@ consist of:
+** @(BPR2a)@ @channel@ string - the name of the channel
+** @(BPR2b)@ @messageId@ string - an optional string containing the @messageId@ prefix for the published message(s), if successful
+** @(BPR2c)@ @error@ @ErrorInfo@ - an optional @ErrorInfo@ indicating the reason the message(s) failed to publish, if unsuccessful
+
+h4. BatchPresenceResult
+
+* @(BGR1)@ Contains information about the result of a batch presence request for a single channel
+* @(BGR2)@ The attributes of @BatchPublishChannelResult@ consist of:
+** @(BGR2a)@ @channel@ string - the name of the channel
+** @(BGR2b)@ @presence@ @PresenceMessage[]@ - an optional array containing all members present on the channel if the request succeeded
+** @(BGR2c)@ @error@ @ErrorInfo@ - an optional @ErrorInfo@ indicating the reason the presence request failed for the given channel, if unsuccessful
+
+h4. TokenRevocationSpecifierTarget
+
+* @(TRT1)@ Describes which tokens should be affected by a token revocation request
+* @(TRT2)@ The attributes of @TokenRevocationTargetSpecifier@ consist of:
+** @(TRT2a)@ @type@ string - either "clientId" or "revocationKey"
```

Comment:
or "channel". (or some other type that ably might add support for in future, so sdks probably shouldn't be trying to do any clientside validation on the type, other than that it's a nonempty string)